### PR TITLE
Update the pure fp16 training usage for ResNet50.

### PR DESCRIPTION
**Pull request by wzzju**
Update the pure fp16 training usage for ResNet50.

* Performance verification

|Batch Size | Pure FP16(without loss scaling) | Pure FP16 (with loss scaling) | 
|---|---|---|
|208 | 1417.59 images/sec| 1407.29 images/sec|

* Convergence verification
	
|Training strategy |Training loss | Testing loss | Training acc1 | Testing acc1 | 
|---|---|---|---|---|
| FP32 | 0.937 | 0.967 | 0.772 | 0.757 | 
| Pure FP16(without loss scaling)  |0.829 | 0.966 |  0.796 | 0.755 | 
| Pure FP16(with loss scaling)  |0.864 | 0.972 |  0.789 | 0.756 |

**The introduction of dynamic loss scaling function will cause some slight performance degradation, but will greatly improve the convergence of the model.**

* Training loss consistency verification during the whole training process.
image

* Average loss and test accuracy consistency verification after each epoch.

image
image
image
image

### CODE REVIEW

**Review by Xreki — APPROVED** [review]
LGTM
